### DISCUSSION TITLE
get losses and metrics of epoch

### DISCUSSION THREAD

**Discussion by yangtzech** [timeseriesAI/tsai#370] [Q&A] [answered]
using `learn.fit_one_cycle()`  can show losses and metrics during the progress of training, and graph below, but is there a way that I can directly get those valuse during training and store them in some variables, like `a = learn.metric['rmse'] or a = learn.losses`

![image](https://user-images.githubusercontent.com/41911522/149873459-5ba595e2-7353-494a-baca-e2bba94e9adf.png)

**Comment by oguiza** [comment, marked as answer]
Hi @yangtzech, 
fastai stores those values by default in an object called learn.recorder. 
You can access the losses using learn.recorder.losses, metric names using learn.recorder.metric_names, and their values using learn.recorder.values.

**Reply by yangtzech** [reply]
Thank you! Problem solved.
